src/main/java/org/jabref/http/dto/SimpleHttpResponse.java

```diff
@@ -0,0 +1,62 @@
+package org.jabref.http.dto;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+
+public record SimpleHttpResponse(int statusCode, String responseBody , String responseMessage) {
+
+    public SimpleHttpResponse(int statusCode,String responseBody,String responseMessage) {
+        this.statusCode=statusCode;
+        this.responseBody=truncateResponseBody(responseBody);
+        this.responseMessage=responseMessage;
+    }
+
+    public SimpleHttpResponse(HttpURLConnection connection) throws IOException {
+        this(
+                connection.getResponseCode(),
+                getResponseBody(connection),
+                connection.getResponseMessage()
+        );
+    }
+
+    /**
+     * Truncates the response body to 1 KB if it exceeds that size.
+     * Appends "... (truncated)" to indicate truncation.
+     *
+     * @param responseBody  the original response body
+     * @return  the truncated response body
+     */
+    private static String truncateResponseBody(String responseBody) {
+        byte[] bytes = responseBody.getBytes(StandardCharsets.UTF_8);
+        int maxLength = 1024; // 1 KB
+        if (bytes.length > maxLength) {
+            // Truncate the response body to 1 KB and append "... (truncated)"
+            return new String(bytes, 0, maxLength, StandardCharsets.UTF_8) + "... (truncated)";
```

Comment:
the three dots would in my understanding expand the 1024 bytes to 1027 bytes. is this expected to happen?

Reply:
yes they will , should i changed the maxLength to 1024 - 15 (length of the "... (truncated)" suffix) or get rid of the suffix